lib/i18n/backend/cache.rb

```diff
@@ -70,11 +77,32 @@ def cache_key_digest=(key_digest)
     def perform_caching?
       !cache_store.nil?
     end
+
+    def cached_keys
+      @@cached_keys
+    end
+
+    def add_cached_key(cache_key)
+      @@cached_keys << cache_key
+    end
+
+    def clear_cache
+      if I18n.perform_caching?
```

Comment:
If we have cache store dedicated for i18n we can simply flash all keys (like Redis with namespace), otherwise we need to clear key by key.